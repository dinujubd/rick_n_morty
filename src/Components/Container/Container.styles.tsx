import { createStyles, makeStyles, Theme } from "@material-ui/core";

export const useStyles = (isMobile: boolean) => makeStyles((theme: Theme) =>
    createStyles({
        root: {
            height: isMobile ? 'auto': '100vh'
        },
        appContainer: {
            height: isMobile ? 'auto': '90vh'
        },
        leftCol: {
            backgroundColor: `${theme.palette.background.paper}7`,
            padding: theme.spacing(2),
            borderRadius: `${theme.shape.borderRadius}px 0 0 ${theme.shape.borderRadius}px`,
            boxShadow: theme.shadows[5]
        },
        charList: { height: '70vh', overflowY: 'auto' },
        charDetails: {
            height: isMobile ? 'auto': '90vh',
            background: theme.palette.background.paper,
            overflow: 'hidden',
            borderRadius: `0 ${theme.shape.borderRadius}px ${theme.shape.borderRadius}px 0`,
            boxShadow: theme.shadows[5]
        },
        genderIcon: {
            marginLeft: theme.spacing(2)
        },
        details: {
            padding: theme.spacing(2)
        },
        logo: {
            position: 'absolute',
            left: '20px',
            top: '20px',
            height: '50px',
            zIndex: -1,
        }
    }),
);