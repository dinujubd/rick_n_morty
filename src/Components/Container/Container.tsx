import { Box, createStyles, Grid, makeStyles, Theme } from "@material-ui/core";
import AcrylicBackgroud from "../AcrylicBackground/AcrylicBackgroud";
import CharacterCard from "../CharacterCard/CharacterCard";
import SearchBar from "../SearchBar/SearchBar";
import CharacterList from '../CharacterList/CharacterList';

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            height: '100vh'
        },
        appContainer: {
            height: '90vh'
        },
        leftCol: {
            background: 'rgba(255,255,255, 0.6)',
            padding: 10,
            borderRadius: '10px 0 0 10px'
        },
        charList: { height: '70vh', overflowY: 'auto' },
        charDetails: { background: 'white', overflow: 'hidden', borderRadius: '0 10px 10px 0' },
        genderIcon: {
            marginLeft: 5
        },
        details: {
            padding: '1em'
        }, logo: {
            position: 'absolute',
            left: '20px',
            top: '20px',
            height: '50px',
            zIndex: -1,
        }
    }),
);

export const Container: React.FC = () => {
    const logoUrl = "https://upload.wikimedia.org/wikipedia/commons/b/b1/Rick_and_Morty.svg";
    const classes = useStyles();

    return (
        <>
            <AcrylicBackgroud />
            <img className={classes.logo} src={logoUrl} />
            <Grid
                container
                className={classes.root}
                justify="center"
                alignItems="center"
            >
                <Grid
                    className={classes.appContainer}
                    container
                    justify="center"
                >
                    <Grid container direction="column" className={classes.leftCol} xs={12} sm={4} lg={4} >
                        <Grid container justify="center">
                            <SearchBar />
                        </Grid>
                        <Box className={classes.charList}>
                            <CharacterList />
                        </Box>
                    </Grid>
                    <Grid className={classes.charDetails} xs={12} sm={5} lg={5} >
                        <CharacterCard />
                    </Grid>
                </Grid>
            </Grid>
        </>
    )
};


Container.displayName = 'Conitaner';

export default Container;