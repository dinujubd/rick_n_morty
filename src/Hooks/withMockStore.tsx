import * as React from 'react';

import configureMockStore from "redux-mock-store";
import { Provider } from 'react-redux';
import mockCharacter from '../Mocks/mockCharacter';


export const withMockStore = (component: React.ReactElement) => {

    const mockStore = configureMockStore();
    const store = mockStore({characters: {
        characters: [{mockCharacter}],
        currentCharacter: mockCharacter
    }});

    return (
        <Provider store={store}>
            {component}
        </Provider>
    );

}