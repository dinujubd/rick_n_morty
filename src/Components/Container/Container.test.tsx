import renderer from 'react-test-renderer';
import { withMockStore } from '../../Hooks/withMockStore';
import { Container } from './Container';

describe('<Container />', () => {
    it('should renders correctly', () => {
        const tree = renderer
            .create(withMockStore(<Container />))
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});