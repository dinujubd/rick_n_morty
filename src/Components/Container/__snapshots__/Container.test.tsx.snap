// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should renders correctly 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-10"
  >
    <div
      className="MuiBox-root MuiBox-root-11 makeStyles-root-9"
    />
  </div>,
  <img
    className="makeStyles-logo-8"
    src="https://upload.wikimedia.org/wikipedia/commons/b/b1/Rick_and_Morty.svg"
  />,
  <div
    className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root makeStyles-appContainer-2 MuiGrid-container MuiGrid-justify-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
      >
        <div
          className="MuiGrid-root makeStyles-leftCol-3 MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
          >
            <form
              className="MuiPaper-root makeStyles-root-12 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <button
                aria-label="menu"
                className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-14"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    onClick={[Function]}
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                    />
                  </svg>
                </span>
              </button>
              <div
                className="MuiInputBase-root makeStyles-input-13"
                onClick={[Function]}
              >
                <input
                  aria-label="Search Character"
                  className="MuiInputBase-input"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Search Character"
                  type="text"
                />
              </div>
              <button
                aria-label="search"
                className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-14"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="submit"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                  </svg>
                </span>
              </button>
            </form>
          </div>
          <div
            className="MuiBox-root MuiBox-root-17 makeStyles-charList-4"
          >
            <ul
              className="MuiList-root MuiList-padding"
            >
              <div
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                <div
                  className="MuiListItemAvatar-root"
                >
                  <div
                    className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiAvatar-fallback"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  className="MuiListItemText-root"
                />
              </div>
            </ul>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root makeStyles-charDetails-5 MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-5"
      >
        <div
          className="MuiPaper-root MuiCard-root makeStyles-root-18 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            className="MuiGrid-root makeStyles-container-19 MuiGrid-container MuiGrid-direction-xs-column"
          >
            <img
              alt="mock name"
              className="makeStyles-cover-20"
            />
            <div
              className="MuiGrid-root makeStyles-details-25 MuiGrid-container MuiGrid-justify-xs-space-between"
            >
              <div
                className="MuiBox-root MuiBox-root-27 makeStyles-nameBlock-23"
              >
                <h2
                  className="MuiTypography-root makeStyles-name-21 MuiTypography-h5"
                >
                  mock name
                </h2>
                <img
                  className="makeStyles-genderIcon-24"
                />
              </div>
              <img
                className="makeStyles-lifeStatus-26"
              />
            </div>
            <div
              className="MuiGrid-root makeStyles-details-25 MuiGrid-container"
            >
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
              >
                <p
                  className="MuiTypography-root makeStyles-dataTitle-22 MuiTypography-body1 MuiTypography-gutterBottom"
                >
                  Species
                </p>
                <p
                  className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
                />
              </div>
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
              >
                <p
                  className="MuiTypography-root makeStyles-dataTitle-22 MuiTypography-body1 MuiTypography-gutterBottom"
                >
                  Location
                </p>
                <p
                  className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
                >
                  mock location
                </p>
              </div>
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
              >
                <p
                  className="MuiTypography-root makeStyles-dataTitle-22 MuiTypography-body1 MuiTypography-gutterBottom"
                >
                  Origin
                </p>
                <p
                  className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
                >
                  mock origin
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
